Content.Shared/Xenoarchaeology/Artifact/XAT/Components/XATCompNearbyComponent.cs

```diff
@@ -0,0 +1,20 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAT.Components;
+
+/// <summary>
+/// This is used a XAT that activates when an entity fulfilling the given whitelist is nearby the artifact.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(XATCompNearbyComponent)), AutoGenerateComponentState]
+public sealed partial class XATCompNearbyComponent : Component
+{
+    [DataField(customTypeSerializer: typeof(ComponentNameSerializer)), AutoNetworkedField]
+    public string Comp = "Item";
```

Comment:
rename to ComponentTypeName? Comp is used in Entity<> type and can thus be misleading!